src/lib/core/CHIPConfig.h

```diff
@@ -813,6 +813,25 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #define CHIP_IM_MAX_NUM_COMMAND_HANDLER 4
 #endif
 
+/**
+ * @def CHIP_IM_MAX_NUM_PATH_PER_SUBSCRIPTION
+ *
+ * @brief The maximum number of path per subscription.
+ */
+#ifndef CHIP_IM_MAX_NUM_PATH_PER_SUBSCRIPTION
+#define CHIP_IM_MAX_NUM_PATH_PER_SUBSCRIPTION 3
```

Comment:
The resource minimas are enforced such that a fabric with only one subscription can consume all 9 read paths for that single subscription.  Furthermore, it is possible (although potentially unwise) for a greedy subscriber to consume even more paths if there are no other fabrics.

This feature doesn't appear to consider this.  I presume this is the reason that this #define is needed now, where it wasn't before.

Reply:
I understand now. I've since revamped the storage API, and the size estimation is done according to CHIP_IM_SERVER_MAX_NUM_PATH_GROUPS_FOR_SUBSCRIPTIONS, and reverted this bit.